containers/unit_tests/TestDynViewAPI.hpp

```diff
@@ -1431,7 +1431,7 @@ class TestDynViewAPI {
     //  Send a std::pair as argument to a rank
     sdView dssp = Kokkos::subdynrankview(
         d7, Kokkos::ALL(), Kokkos::ALL(), Kokkos::ALL(), Kokkos::ALL(),
-        Kokkos::ALL(), Kokkos::ALL(), std::pair<unsigned, unsigned>(1, 2));
+        Kokkos::ALL(), Kokkos::ALL(), Kokkos::pair<unsigned, unsigned>(1, 2));
```

Comment:
This is still supposed to work with std::pair as long as you are on the host. Why does it need to change?

Reply:
It works now, but "as long as you're on the host" is a complicated question when you're dealing with `__host___ __device__` function templates (and remember, those can't be "just warnings" because of the way clang works).  I'm okay with switching this test back at some point, but before now we were calling a `__host__ __device__` function with a `__host__`-only type (`std::pair`), which was a bug.

Reply:
It is not a bug to call a `__host__ __device__` function with a `__host__` only type. It is a bug to compile a `__host__ __device__` function called with a `__host__` type for the device. For inlined or templated functions that means this is fine, since it will only be compiled for the scope in which it was invoked.